docs/src/configuration.md

```diff
@@ -81,35 +102,59 @@ The following implementations should work:
-configuration file located at `~/.julia/prefs/MPI.toml` and so can be used for subsequent
-MPI builds.
+As an example you can use [`MPI.use_system_binary(;export_prefs)`](@ref) to create a file `Project.toml` containing:
 
-- `JULIA_MPI_BINARY`: can be set to either the empty string (to use the default implementations
-  above) or `system` (to use a system-provided implementation).
-- `JULIA_MPI_PATH`: the top-level installation directory of MPI. i.e. the library should
-  be located in `${JULIA_MPI_PATH}/lib` and `mpiexec` in `${JULIA_MPI_PATH}/bin`
-- `JULIA_MPI_LIBRARY`: the library name or full path of the MPI shared library. By
-  default, it will attempt to look for common MPI library names in the standard library
-  paths (e.g. `libmpi`, `libmpich`, `msmpi`).
-- `JULIA_MPIEXEC`: the name (or full path) of the MPI launcher executable. The default is
-  `mpiexec`, but some clusters require using the scheduler launcher interface (e.g. `srun`
-  on Slurm, `aprun` on PBS).
-- `JULIA_MPIEXEC_ARGS`: Additional arguments to be passed to MPI launcher.
+```
+[extras]
+MPIPreferences = "3da0fdf6-3ccc-4f1b-acd9-58baa6c99267"
+
+[preferences.MPIPreferences]
+abi = "OpenMPI"
+binary = "system"
+libmpi = "/software/mpi/lib/libmpi.so"
+mpiexec = "/software/mpi/bin/mpiexec"
+```
+copying that to a central location like: `/software/mpi/julia` and setting `JULIA_LOAD_PATH=":/software/mpi/julia"` (note the `:` before the path) in the coressponding module file (preferrably the MPI installation, or the Julia module)
+, will cause the user to default to your cluster MPI installation.
 
-The following variables are also queried:
+The user can still create a differing per Julia project that
+will take precedent.
 
-- `JULIA_MPI_INCLUDE_PATH`: the directory containing the MPI header files.
-- `JULIA_MPI_CFLAGS`: C flags passed to the constant generation build (default: `-lmpi`)
-- `JULIA_MPICC`: MPI C compiler (default: `mpicc`)
+### Using a different JLL provided MPI library
+
+The following JLL implementations are provided:
+
+- `MicrosoftMPI_jll`: Default for Windows
+- `MPICH_jll`: Default on all Unix systems
+- [`MPItrampoline_jll`](https://github.com/eschnett/MPItrampoline): Binaries built against MPItrampoline can be efficiently retargetted to a system MPI implementation.
+- `OpenMPI_jll`
+
+```@doc
+MPI.use_jll_binary
+```
 
+## Environment variables for the testsuite
 The test suite can also be modified by the following variables:
 
-- `JULIA_MPIEXEC_TEST_ARGS`: Additional arguments to be passed to the MPI launcher for the tests only.
+- `JULIA_MPI_TEST_NPROCS`: How many ranks to use within the tests
 - `JULIA_MPI_TEST_ARRAYTYPE`: Set to `CuArray` to test the CUDA-aware interface with
   [`CUDA.CuArray](https://github.com/JuliaGPU/CUDA.jl) buffers.
+- `JULIA_MPI_TEST_BINARY`: Check that the correct MPI binary is used for the tests
+- `JULIA_MPI_TEST_ABI`: Check that the correct MPI ABI is used for the tests
```

Comment:
From the docs it is unclear what I need to do with these variables. Set them to `true`?

Reply:
Need to be set to what you expect the variable to be.

E.g. `JULIA_MPI_TEST_BINARY=binary` and the test will fail if you are using a jll

Reply:
```suggestion
- `JULIA_MPI_TEST_BINARY`: Check that the specified MPI binary is used for the tests
- `JULIA_MPI_TEST_ABI`: Check that the specified MPI ABI is used for the tests
```
Maybe like this?